artemis-commons/src/main/java/org/apache/activemq/artemis/utils/ArtemisSystemProperties.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.activemq.artemis.utils;
+
+import java.net.URL;
+import java.util.Properties;
+
+import org.jboss.logging.Logger;
+
+public final class ArtemisSystemProperties {
+
+   private static Logger logger = Logger.getLogger(ArtemisSystemProperties.class);
+
+   static {
+      init();
+   }
+
+   private static ArtemisSystemProperties theInstance;
+
+   private Properties properties;
+   private boolean enable1XPrefixes = false;
```

Comment:
this be marked static final and the loading done all in static method.

Reply:
@michaelandrepearce The class is singleton. 

the reason I did not make it final was because of testcases where I can call init() to reload properties.

Reply:
Though this is being got on hot path later in message buffers, which is not so good. Having it static final will minimise perf impact

Reply:
Actually.. this is not checked on the hot path. 
the one on the ClientSessionImpl is.

Reply:
I tried marking the bits that i think are hot paths. Using phone atm so not easiest